### DISCUSSION TITLE
Ionospheric corrections in ALOS from stackStripMap

### DISCUSSION THREAD

**Discussion by xoanportela** [isce-framework/isce2#599] [Q&A]
Hello everyone,
I'm working with a stack of ALOS PALSAR data in Central America, and I have some issues while correcting the estimated dispersive phase with stackStripMap.py (which I used with the "Ionosphere" workflow). I am not sure if I am properly applying the correction.

After stackStripMap processing, the following outputs are written in each of the ./Ionosphere/_date1_date2_/ folder (excluding the xml and vrt files):
- dJumps.bil
- highBandShelve
- highBand.Sigma
- iono.bil
- iono.bil.filt
- iono.bil.filt.rsc
- iono.bil.sig
- iono.bil.sig.filt
- iono.bil.sig.filt.rsc
- iono.bil.unwCor
- iono.bil.unwCor.filt
- jumps.bil
- lowBandShelve
- lowBand.Sigma
- mask.bil
- mJumps.bil
- nonDispersive.bil
- nonDispersive.bil.filt
- nonDispersive.bil.sig
- nonDispersive.bil.sig.filt
- nonDispersive.bil.unwCor
- nonDispersive.bil.unwCor.filt

I understand the the **iono.bil(.filt)** contains the dispersive phase to be subtracted from the unwrapped interferograms. Is this correct? So then, the **iono.bil.sig.filt** and the **iono.bil.unwCor.filt** contain the standard deviation and coherence respectively? Or should I use the **nonDispersive.bil.filt** file directly?

I am using imageMath.py to correct the interferograms, but I am not entirely sure whether I am doing it correctly. Here is an example for one pair:

imageMath.py -e='a_0;a_1-b_0' -s BIL --a=Igrams/20071122_20081124/filt_20071122_20081124_snaphu.unw --b=Ionosphere/20071122_20081124/iono.bil.filt  -o Igrams/20071122_20081124/filt_topophase_nondispersive.unw

After doing this (recursively), I obtain some seemingly good corrections (removing a ramp-like ionospheric phase), but others seem to introduce some kind of artifact. I attach an example for five pairs where this can be seen.
However, the **iono.bil.unwCor.filt** files look very strange (almost like a mask). I have produced these figures with Mintpy. The example is from a Descending track.

![Picture1](https://user-images.githubusercontent.com/111895403/195460938-7d47aadf-6dde-4a0f-8cc6-92a8d17f17c5.png)

Is this artifact caused by a bad estimation of the dispersive phase?

I hope someone can help me clarifying this issues.
Thank you very much in advance!
Juan

**Comment by remotesensingvn** [comment]
I have the same interest with ionospheric correction in stripmapstack. Can you change the ionospheric estimation to the same range as interferogram and how about the low and high band interferogram for one of the problematic pair?

**Comment by xoanportela** [comment]
Hello,
sorry for the late reply. So I think the problem seems to appear in interferograms with low coherence.
![Picture2](https://user-images.githubusercontent.com/111895403/196060188-5c992bad-6d18-4a8f-b045-4c31ed869455.png)


I show here the corrections for two of the interferograms (in the same scale). The masks generated in the ionospheric estimation seem to be very different. While in the coherent interferogram (20071122_20080222) this mask removes water bodies, in the incoherent one it removes most of the interferogram. So I guess this mask indicates where the ionospheric estimation is reliable?
I didn't plot the masked corrections in the second interferogram (20071122_20081124) so it's easier to see the patterns.
Note also that I am assuming that the estimated dispersive phase is that of the **_iono.bil.filt_** file, is this correct?
![Picture3](https://user-images.githubusercontent.com/111895403/196060208-c7b06d73-bc9a-407f-9851-c087d836c26c.png)


Also, there doesn't seem to be any problem with the Low and High band interferograms (which look very similar, as expected).
![Picture4-min](https://user-images.githubusercontent.com/111895403/196061510-19e8c8cd-0ffb-44dc-b232-6b0e4825ab36.png)

**Reply by remotesensingvn** [reply]
I think you are having good conclusions. However, I'm facing the issue even with relatively good coherence pairs. My study area is large so the ionosphere correction does not perform well. Unfortunately, JPL did not provide training or details about stack processing, especially, ionosphere correction in stack.

**Reply by Jackbakker4444** [reply]
@remotesensingvn, Did you eventually find a solution to this issue? I am facing the same problems with my thesis research.
